### PR TITLE
Use JSON format for commands signature

### PR DESCRIPTION
**Pull request by merwanehamadi**
### Changes

Use more formal definition of command arguments, using JSON, so that we can prepare the OpenAI functions.
### Documentation

- changed signature on all commands
- removed git_operations, dowload_file and replace_in_file
- Fixed summarization using the wrong method to check if valid JSON
### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes. 
- [x] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports --ignore-pass-after-docstring autogpt tests --in-place
    ```

### CODE REVIEW

**Review by erik-megarad — APPROVED** [review]

File: autogpt/command_decorator.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 def command(
     name: str,
     description: str,
-    signature: str,
+    arguments: dict = None,
```
Comment (Pwuts) [inline autogpt/command_decorator.py]: * `dict` is nonspecific
* `x: dict = None` gives a type error

File: autogpt/commands/execute_code.py [hunk 2]
```diff
@@ -20,7 +20,18 @@
...
+    {
+        "code": {
+            "type": "string",
+            "description": "The Python code to run",
+            "required": True,
+        },
+        "name": {
+            "required": True,
+            "type": "string",
+            "description": "A name to be given to the python file",
```
Comment (Pwuts) [inline autogpt/commands/execute_code.py]: nit: inconsistent order of properties

File: autogpt/commands/file_operations.py [hunk 3]
```diff
@@ -307,7 +287,7 @@ def delete_file(filename: str, agent: Agent) -> str:
         return f"Error: {err}"
 
 
-@command("list_files", "List Files in Directory", '"directory": "<directory>"')
+@command("list_files", "List Files in Directory", {})
```
Comment (Pwuts) [inline autogpt/commands/file_operations.py]: This doesn't seem right

File: autogpt/commands/image_gen.py
Comment (Pwuts) [inline autogpt/commands/image_gen.py:1]: I still don't agree with removing this

File: autogpt/commands/git_operations.py
Comment (Pwuts) [inline autogpt/commands/git_operations.py:1]: Why must this be removed?

File: autogpt/commands/web_search.py [hunk 4]
```diff
@@ -14,12 +14,17 @@
 
 
 @command(
-    "google",
-    "Google Search",
-    '"query": "<query>"',
-    lambda config: not config.google_api_key,
+    "web_search",
+    "Web Search",
```
Comment (Pwuts) [inline autogpt/commands/web_search.py]: "Search the web"?

File: autogpt/models/command.py [hunk 5]
```diff
@@ -17,7 +17,7 @@ def __init__(
         name: str,
         description: str,
         method: Callable[..., Any],
-        signature: str = "",
+        signature: dict = None,
```
Comment (Pwuts) [inline autogpt/models/command.py]: Same comment as earlier, this is poor typing

**Review by Pwuts — CHANGES_REQUESTED** [review]
Please link back to the original PR and give @erik-megarad some credit in the description :)

**Comment by merwanehamadi** [comment]
thanks @erik-megarad :
https://github.com/Significant-Gravitas/Auto-GPT/pull/4683/files

**Review by richbeales — APPROVED** [review]

**Pull request merged**